internal/controller/resource_manager.go

```diff
@@ -43,18 +90,18 @@ type ResourceManager struct {
 	limiter *rate.Limiter
 }
 
-// NewResourceManager creates a new ResourceManager for a specific resource type.
+// New creates a ResourceManager from the config.
 // The logger will automatically include the resource name in all log messages.
-func NewResourceManager(config ResourceManagerConfig) *ResourceManager {
+func (c ResourceManagerConfig) New() *ResourceManager {
```

Comment:
OK with making this a method on `NewResourceManager`.
But why rename this into `New`?
`configblah.New()` doesn't make sense to me.